consent/src/main/java/com/seaplain/android/consent/Consent.java

```diff
@@ -64,6 +64,12 @@ private void onRequestPermissionsResult(int requestCode, @NonNull String[] permi
         PermissionRequest request = mPendingRequests.get(requestCode);
         if (request != null) {
             mPendingRequests.remove(requestCode);
+
+            if (permissions.length == 0 && grantResults.length == 0) {
+                // According to Activity#onRequestPermissionsResult, this means permissions request was cancelled.
```

Comment:
The requester would probably want to know about this. IMO this is akin to declining the permission and thus should invoke that callback.

Reply:
I don't think the callback should be invoked. The documentation says this should be treated as a cancellation and that's not the same thing as declining the permission.

Google's lib for permissions for example doesn't invoke any callbacks on cancellation: https://github.com/googlesamples/easypermissions/blob/master/easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java#L251

This makes sense because the things that are usually done when a permission is declined e.g. disable a feature or display a message notifying that app will have limited behaviour should not be done on a cancellation because the user is bound to trigger the permission request again and only then will they decide whether to grant the permission.